nrf70_bm_lib/docs/source/nrf70_bm_porting_guide.rst

```diff
@@ -12,19 +12,79 @@ Zephyr functionality used in the reference implementation:
 
-* *OS primitives*: The reference implementation uses the following Zephyr OS primitives:
+.. list-table:: OS primitives porting guidelines
+  :header-rows: 1
+
+  * - OS primitive
+    - Description
+    - OSAL API
+    - Reference Zephyr API
+  * - Tasklet
+    - Used to offload tasks from the ISR context to the main task context.
+    - tasklet_schedule()
+    - k_work_submit()
+  * - Heap
+    - Used to allocate memory for the nRF70 Series driver.
+    - mem_alloc()
+    - k_malloc()
+  * - Lists
+    - Used to manage the list of tasks to be done in the context of the nRF70 Series.
+    - llist_***
+    - sys_slist_***
+  * - Sleep/Delay
+    - Used to manage the sleep and delay for the nRF70 Series.
+    - sleep_ms()/delay_us()
+    - k_msleep()/k_usleep()
+  * - Spinlocks
+    - Used to manage the synchronization between the nRF70 Series and the host MCU.
+    - spinlock_***
+    - k_sem_***
+  * - Timers
+    - Used to manage the timers for the nRF70 Series driver, esp. for low power mode.
+    - timer_***
+    - k_work_delayable_***
+
+
+.. note ::
+
+   The synchronization primitives used in the latest reference implementation have been updated to use Zephyr spinlocks instead of semaphores.
 
-    - Workqueue: Used to schedule work or delayed work to be done.
-    - Heap: Used to allocate memory for the nRF70 Series.
-    - Lists: Used to manage the list of tasks to be done in the context of the nRF70 Series.
-    - Sleep/Delay: Used to manage the sleep and delay for the nRF70 Series.
-    - Semaphore: Used to manage the synchronization between the nRF70 Series and the host MCU.
 * *Driver model*: The reference implementation uses the Zephyr driver model to manage the nRF70 Series device.
 
-    - SPI: Uses Zephyr's SPI driver to communicate with the nRF70 Series over SPI.
-    - GPIO: Used to manage the host IRQ GPIO pin of the nRF70 Series.
+    .. list-table:: Driver model porting guidelines
+      :header-rows: 1
+
+      * - Driver model component
+        - Description
+        - OSAL API
+        - Reference Zephyr API
+      * - SPI
+        - Uses Zephyr's SPI driver to communicate with the nRF70 Series over SPI.
+        - qspi_*_reg32()/qspi_cpy_*()
+        - spi_transceive_dt()
+      * - GPIO
+        - Uses Zephyr's GPIO driver to manage the GPIO pins of the nRF70 Series.
+        - bus_qspi_dev_add()/bus_qspi_dev_rem()
+        - gpio_pin_configure_dt()
 
 * *Build time tools*: The reference implementation uses the following build-time tools:
 
-    - DTS: Used to define the GPIO pins of the nRF70 Series.
-    - Kconfig: Used to define the configuration options for the nRF70 Series driver.
+    - DTS: Used to define the GPIO pins of the nRF70 Series
+
+        - the nRF70 Series BM driver uses C headers over DTS for any hardware-specific configuration
```

Comment:
So far I only saw the two TX power ceil headers - is there more?
With respect to GPIOs - it would be nice if we can point on the auto-generated DTS headers that point to the GPIO configuration.

Reply:
I will add a section for picking up auto-generated files from the Zephyr build.